components/Boards/PlayerVsPlayer.js

```diff
@@ -0,0 +1,112 @@
+import { Chessboard } from "react-chessboard";
+import Chess from "chess.js";
+import { useState, useEffect } from "react";
+import MoveInfo from "../MoveInfo";
+import PlayerNameDisplay from "../PlayerNameDisplay";
+import GameTerminal from "../GameTerminal";
+
+export default function RandomMoveEngine() {
+  const [game, setGame] = useState(null);
+  const [fen, setFen] = useState(game?.fen());
+  const [previousMove, setPreviousMove] = useState(null);
+  const [moveStatus, setMoveStatus] = useState({});
+  const [history, setHistory] = useState([]);
+
+  const latestHistory = history[history.length - 1];
+
+  //temporarily static player-names
+  const playerName = "Player One";
+  const oppenentName = "Player Two";
+
+  // // ---CREATE GAME OBJECT---
+  useEffect(() => {
+    setGame(new Chess());
+  }, []);
+
+  // // ---CREATE SEPERATE GAME HISTORY TO BYPASS GAME RESETS---
+  function historyStorage() {
+    if (
+      latestHistory?.color !==
+      game?.history({ verbose: true })[
+        game.history({ verbose: true }).length - 1
+      ]?.color
```

Comment:
do you need optional chaining here?

will `historyStory()` ever be called if there is no game? It looks like it is only called in your makeAMove function and there you work with a copy of game, so game should also be defined when historyStory() is called.

Reply:
I´ll get an undefined, if I don´t use optional chaining:

![Screenshot 2023-05-11 at 10 08 28](https://github.com/marte1337/capstone-project/assets/54943747/d51ff61d-1914-43ef-a90e-13454600e465)
![Screenshot 2023-05-11 at 10 08 43](https://github.com/marte1337/capstone-project/assets/54943747/6ed37e06-6151-4956-b851-45aa2801a30b)

`historyStorage()` is being called inside makeAMove and also inside the "create-zombie" useEffect. This is important, because inside create zombie the game history gets a reset but the historyStorage() saves the old history and keeps collecting.

Reply:
Alright, then you apparently need the optional chaining there and should keep it